pkg/config/setup/config_nix.go

```diff
@@ -15,9 +15,9 @@ var (
 	// It might be overridden at build time
 	InstallPath = "/opt/datadog-agent"
 
-	// defaultRunPath is the default path for the agent's runtime files
-	// It might be overridden at build time
-	defaultRunPath = "/opt/datadog-agent/run"
+	// defaultRunPath is the default run path
+	// It is set in osinit to take into account InstallPath overrides
+	defaultRunPath = ""
```

Comment:
This [commit](https://github.com/DataDog/datadog-agent/commit/3cc76cdbbc05b7d001a485082e98a8f13d75eb2d) changed from `var defaultRunPath = filepath.Join(InstallPath, "run")` to `var defaultRunPath = "/opt/datadog-agent/run"`

Is there a reason to not revert the change?
Your change is not strictly equivalent because `defaultRunPath` can be access before it is initialized. See 
https://go.dev/play/p/nRoHIouGssW. 
Also using `var defaultRunPath = "/opt/datadog-agent/run"` is more simple.

Reply:
Only constants can be overridden at build time unfortunately so it's not strictly equivalent. This other PR made the value a constant in order to allow overrides for the installer binary

Reply:
What do you think about using something like `defaultRunPath = "NOT_INITIALIZED"`. At least wrong usage would lead to a better error?